src/components/MenuPanel/MenuPanel.js

```diff
@@ -1,119 +1,66 @@
-  constructor(props: Props) {
-    super(props)
-
-    this.state = {
-      isActionsMoreActive: false,
-      isWalletManagerActive: false,
-    }
-  }
-
-  toggleActionsMore = () => {
-    this.setState({
-      isWalletManagerActive: false,
-      isActionsMoreActive: !this.state.isActionsMoreActive,
-    })
-  }
-
-  toggleWalletManager = () => {
-    this.setState({
-      isActionsMoreActive: false,
-      isWalletManagerActive: !this.state.isWalletManagerActive,
-    })
+export class MenuPanel extends PureComponent<Props> {
+  static defaultProps = {
+    isMnemonic: false,
+    isMinimized: false,
   }
 
   render() {
     const {
-      getMoreAddresses,
-      setActiveAddress,
-      items,
-      addresses,
-      addressNames,
+      walletName,
       fiatCurrency,
-      activeWalletId,
+      mnemonicAddressName,
+      previousRouteNameFallback,
       fiatBalance,
+      isMnemonic,
+      isMinimized,
     }: Props = this.props
 
-    try {
-      const wallet: Wallet = getWallet(items, activeWalletId)
-
-      const {
-        type,
-        isReadOnly,
-      }: Wallet = wallet
-
-      const {
-        isActionsMoreActive,
-        isWalletManagerActive,
-      }: ComponentState = this.state
-
-      return (
-        <div className={classNames('menu-panel', isReadOnly && '-read-only')}>
-          <div className='top'>
-            <div className='logo'>
-              <JLogo isOnlyIcon />
-            </div>
-            <div className='separator' />
-            <div className='ticker'>
-              <MenuPanelBalanceTicker balance={fiatBalance} currency={fiatCurrency} />
-            </div>
-          </div>
-          <div className='actions'>
-            <MenuPanelMainAction
-              type={type}
-              isReadOnly={isReadOnly}
-            />
-            <MenuPanelActions
-              toggle={this.toggleActionsMore}
-              isActive={isActionsMoreActive}
-              activeWalletId={activeWalletId}
-            />
-          </div>
-          <MenuPanelWalletManager
-            toggle={this.toggleWalletManager}
-            getMoreAddresses={getMoreAddresses}
-            setActiveAddress={setActiveAddress}
-            items={items}
-            addresses={addresses}
-            addressNames={addressNames}
-            activeWalletId={activeWalletId}
-            isActive={isWalletManagerActive}
-          />
-        </div>
-      )
-    } catch (err) {
-      return null
-    }
+    return (
+      <header
+        className={classNames(
+          '__menu-panel',
+          menuPanelStyle.core,
+          isMinimized && menuPanelStyle.minimized,
+        )}
+      >
+        <Top
+          walletName={walletName}
+          fiatCurrency={fiatCurrency}
+          mnemonicAddressName={mnemonicAddressName}
+          fiatBalance={fiatBalance}
+          isMnemonic={isMnemonic}
+        />
+        <Separator />
+        <Actions />
+        <Separator />
```

Comment:
Why do you put separator as separate component? It is purely decorative entity not really useful by itself, only as part of something (like `Top`, or `Actions`, or `Settings`) and can be even done without using any HTML (`:after` of `border`)